### PR TITLE
feat: require argument for config export to include urls

### PR DESCRIPTION
**Pull request by iProdigy**
* `::dinkexport` - all settings excluding webhook urls
* `::dinkexport all` - all settings including webhook urls
* `::dinkexport webhooks` - only export webhook urls
* `::dinkexport sectionName` - only export notifier settings for a specific section (without url). For example: `::dinkexport bagambles`

### CODE REVIEW

**Comment by iProdigy** [comment]
https://github.com/pajlads/DinkPlugin/pull/160/commits/f675c0ac54460263415c6e53e6bf92bb6db173ae may be too dank, so feel free to merge without it

File: src/main/java/dinkplugin/SettingsManager.java [hunk 1]
```diff
@@ -70,16 +81,52 @@ public boolean isNamePermitted(String name) {
...
+            String[] args = event.getArguments();
+
+            Predicate<String> includeKey;
+            if (args == null || args.length == 0) {
+                includeKey = k -> !WEBHOOK_CONFIG_KEYS.contains(k);
+            } else {
+                includeKey = k -> false;
+
+                for (String arg : args) {
+                    if ("all".equalsIgnoreCase(arg)) {
```
Comment (pajlada) [inline src/main/java/dinkplugin/SettingsManager.java:112]: Can we add an option for "default" which is the same as sending no argument?

File: README.md [hunk 2]
```diff
@@ -45,15 +45,16 @@ Some notifiers require in-game settings to be configured to send chat messages u
...
 ## Chat Commands
 
-### Export Current Configuration via `::dinkexport`
+### Export Current Configuration via `::DinkExport`
 
 Dink allows you to export your current plugin configuration to the clipboard via the `::dinkexport` chat command.
 
-This export includes *every* setting from individual notifiers to the webhook URLs.
+This export includes settings across all of the notifiers. If you want to include webhook URLs in the export, run the `::dinkexport all` chat command.
+Alternatively, if you *only* want to export the webhook URLs, run the `::dinkexport webhooks` chat command.
```
Comment (pajlada) [inline README.md:53]:
```suggestion
Alternatively, if you *only* want to export the webhook URLs, run the `::dinkexport webhooks` chat command.

You can combine arguments to export specific sets of settings.

**Export Slayer & BA Gambles**: `::dinkexport bagambles slayer`  
**Export webhook overrides only**: `::dinkexport webhookoverrides`  
**Export webhooks & levels notifier**: `::dinkexport webhooks levels`

#### Available selectors:
 - `all` Export everything
 - `webhooks` Primary Webhook URLs & Webhook Overrides
 - `default` Export everything except webhooks
 - `collectionlog`, `pet`, `levels`, `loot` etc to export one notifier's settings (Section header with spaces removed)
```

**Review by pajlada — CHANGES_REQUESTED** [review]
document the multi-selecting

File: src/main/java/dinkplugin/SettingsManager.java (hunk 1 above)
Comment (iProdigy) [reply src/main/java/dinkplugin/SettingsManager.java:112]: Sure, as along as people don't interpret that as exporting the plugin's default values for each config item

Comment (Felanbird) [reply src/main/java/dinkplugin/SettingsManager.java:112]: I do agree this will likely cause confusion that it's not the settings `default` value

Comment (iProdigy) [reply src/main/java/dinkplugin/SettingsManager.java:112]: also maybe we don't need this until a later date where you can do default + specific url

right now:

* export default + notifier => just do export default
* export default => just do export without arg
* export default + webhooks => just do export all

Comment (pajlada) [reply src/main/java/dinkplugin/SettingsManager.java:112]: That's fine by me

**Review by pajlada — APPROVED** [review]

**Review by Felanbird — APPROVED** [review]

**Pull request merged**